### ISSUE TITLE
Golos power delegation for voting and bandwidth

### ISSUE THREAD

**Issue by Nemo1369** [GolosChain/golos#101]
The idea behind Steem Power Delegation is to enable new account creation without having to worry about hackers draining an account signup pool. This process also enables 3rd party websites to create new accounts.

An account can delegate their Steem Power to another account which will give that account extra bandwidth and extra voting power. The amount delegated will be denominated in MVESTS and will automatically be retracted in the event that the source account runs out of MVESTS due to power down. The oldest delegated MVEST will be retracted first.

This feature will enable people with multiple accounts to delegate all of their Steem Power to a single account and reduce unnecessary voting spam.

New Operation:

update_vote_delegation from to MVEST

Accounts may not delegate delegated MVESTS, only their own. All accounts will now have the following fields:

```
vesting_shares // same as today
delegated_vesting_shares // total this account has delegated to others <= vesting_shares
effective_vesting_shares  // vesting_shares - delegated_vesting_shares + total_received
```

This change will not impact how witness scheduling works because witness scheduling already supports its own delegation system.

When Steem Power is delegated curation rewards do not propagate back to the source, but belong to the account that received delegation.

The minimum delegation amount will be the account creation fee or what ever higher value witnesses set by soft fork.

When removing delegated SteemPower, it needs to be held for the VOTE regeneration period (5 business days)... this would still allow someone to vote for the same post twice from two accounts, so the delay needs to be the POST PAYOUT PERIOD. (7 days).

Delegated bandwidth also needs a delay when it is "revoked" to prevent someone from spamming the network by consuming all bandwidth, re-delegate, consume all bandwidth. Might as well make this 7 days as well.

Once we make reward payouts explicit virtual ops, only nodes processing votes or bandwidth calculations would need to perform these calculations.

All nodes participating in the P2P network need to process bandwidth calculations to prevent network spam of pending transactions.

Inspired by https://github.com/steemit/steem/issues/818 and supported by community.

Related to #62

**Comment by Nemo1369** [comment]
Closed by 1186c01

**Comment by Nemo1369** [comment]
Closed by 46aca4d
